src/main/java/com/impactupgrade/nucleus/service/segment/SendGridEmailService.java

```diff
@@ -44,16 +45,18 @@ public void sendEmailTemplate(String subject, String template, Map<String, Objec
     request.setMethod(Method.POST);
     request.setEndpoint("/mail/send");
 
-    Mail mail = new Mail();
-
-    for (String to : tos) {
-      Personalization personalization = new Personalization();
-      personalization.addTo(new Email(to));
-      for (Map.Entry<String, Object> d : data.entrySet()) {
-        personalization.addDynamicTemplateData(d.getKey(), d.getValue());
-      }
-      mail.addPersonalization(personalization);
+    Personalization personalization = new Personalization();
+    tos.forEach(to -> personalization.addTo(new Email(to)));
+    if (CollectionUtils.isNotEmpty(ccs)) {
```

Comment:
Technically don't need the empty check, since `forEach` will simply not do anything if it's empty.

Reply:
The idea was to check both null / empty list. Will replace with != null.